ui/pages/confirmations/components/confirm/info/approve/edit-spending-cap-modal/edit-spending-cap-modal.tsx

```diff
@@ -0,0 +1,168 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { BigNumber } from 'bignumber.js';
+import React, { useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { hexToDecimal } from '../../../../../../../../shared/modules/conversion.utils';
+import {
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  TextField,
+  TextFieldType,
+} from '../../../../../../../components/component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import {
+  estimateGas,
+  updateEditableParams,
+} from '../../../../../../../store/actions';
+import { getCustomTxParamsData } from '../../../../../confirm-approve/confirm-approve.util';
+import { useConfirmContext } from '../../../../../context/confirm';
+import { useAssetDetails } from '../../../../../hooks/useAssetDetails';
+import { useApproveTokenSimulation } from '../hooks/use-approve-token-simulation';
+
+export const getAccountBalance = (userBalance: string, decimals: string) =>
+  new BigNumber(userBalance)
+    .dividedBy(new BigNumber(10).pow(Number(decimals)))
+    .toNumber();
```

Comment:
we can use `calcTokenAmount` where `getAccountBalance` is called.

we should also keep the value as a string instead of a number because converting to numbers may result in a loss of precision when the numbers are greater than Number.MAX_SAFE_INTEGER

. e.g.

```
> const bn = new BigNumber('1000000000000000128');
> bn.toFixed(): '1000000000000000128'
> bn.toNumber():  1000000000000000100
```